python/pyspark/sql/tests.py

```diff
@@ -4331,13 +4354,22 @@ def test_createDataFrame_fallback_enabled(self):
                     self.assertEqual(df.collect(), [Row(a={u'a': 1})])
 
     def test_createDataFrame_fallback_disabled(self):
+        from distutils.version import LooseVersion
         import pandas as pd
+        import pyarrow as pa
 
         with QuietTest(self.sc):
             with self.assertRaisesRegexp(TypeError, 'Unsupported type'):
                 self.spark.createDataFrame(
                     pd.DataFrame([[{u'a': 1}]]), "a: map<string, int>")
 
+        # TODO: remove BinaryType check once minimum pyarrow version is 0.10.0
+        if LooseVersion(pa.__version__) < LooseVersion("0.10.0"):
+            with QuietTest(self.sc):
+                with self.assertRaisesRegexp(TypeError, 'Unsupported type.*BinaryType'):
+                    self.spark.createDataFrame(
+                        pd.DataFrame([[{'a': b'aaa'}]]), "a: binary")
```

Comment:
@BryanCutler, BTW, what kind of data can be binary? `bytearray` and `byte` (`str` in Python 2)?

Reply:
In general, Spark only accepts `bytearray` and pyarrow can accept `bytearray` or `bytes`. In pyarrow < 0.10.0 bytearrays weren't supported though, which is why I used `bytes` for the pandas pdf here.